pages/dao/[symbol]/proposal/components/instructions/UXDProtocol/Redeem.tsx

```diff
@@ -0,0 +1,235 @@
+import { useContext, useEffect, useState } from 'react'
+import * as yup from 'yup'
+import {
+  Governance,
+  ProgramAccount,
+  serializeInstructionToBase64,
+} from '@solana/spl-governance'
+import { USDC } from '@uxd-protocol/uxd-client'
+import Input from '@components/inputs/Input'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import {
+  DEPOSITORY_TYPES,
+  getDepositoryMintSymbols,
+  getDepositoryTypes,
+  redeemUXDIx,
+} from '@tools/sdk/uxdProtocol'
+import { isFormValid } from '@utils/formValidation'
+import {
+  UiInstruction,
+  UXDRedeemForm,
+} from '@utils/uiTypes/proposalCreationTypes'
+import useWalletStore from 'stores/useWalletStore'
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+import { PublicKey, TransactionInstruction } from '@solana/web3.js'
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  Token,
+  TOKEN_PROGRAM_ID,
+} from '@solana/spl-token'
+import { findATAAddrSync } from '@utils/ataTools'
+import Select from '@components/inputs/Select'
+import SelectOptionList from '../../SelectOptionList'
+
+async function checkInitTokenAccount(
+  account: PublicKey,
+  instructions: TransactionInstruction[],
+  connection: any,
```

Comment:
Should not be any, also we could put this function in utils maybe? it's used at multiple places